### PR TITLE
feat: added bitwise coverage solidity example contract (#519)

### LINKED ISSUE #519: [Solidity Assembly Yul support] Bitwise coverage
### Problem

The Hedera network supports smart contracts as part of EVM equivalence.
Occasionally items have shown up that surfaced inequivalence at the language or opcode layer.

Solidity supports inline assembly usage of the Yul language.
However, it may not be clear what coverage the Hedera Smart Contract Service 2.0 has for them

### Solution

- Add solidity contract example that utilize bitwise-coverage options for assembly using Yul

- Add a contract under contracts/yul/bitwise-coverage and tests under respective directories

Ref: https://docs.soliditylang.org/en/latest/yul.html#evm-dialect

### Alternatives

_No response_

```[tasklist]
### Tasks
- [x] not(x)
- [x] and(x, y)
- [x] or(x, y)
- [x] xor(x, y)
- [x] byte(n, x)
- [x] shl(x, y)
- [x] shr(x, y)
- [x] sar(x, y)
```

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:
- added Bitwise contract to cover below opcodes
  - [x] not(x)
  - [x] and(x, y)
  - [x] or(x, y)
  - [x] xor(x, y)
  - [x] byte(n, x)
  - [x] shl(x, y)
  - [x] shr(x, y)
  - [x] sar(x, y)
- added unit tests to cover Bitwise cotract

**Related issue(s)**: #519

Fixes #519

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by ebadiere — COMMENTED** [review]
Nit: Blank line at the end of the file.  Otherwise LG.

File: test/yul/bitwise-coverage/Bitwise.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+
+describe('@solidityequiv5 Bitwise Tests', () => {
```
Comment (AlfredoG87) [inline test/yul/bitwise-coverage/Bitwise.js:24]: how are we deciding what number to put after the `solidityequiv`?

**Review by AlfredoG87 — COMMENTED** [review]
Only the Nit that Eric left and a Question.
Looks good!

File: test/yul/bitwise-coverage/Bitwise.js (hunk 1 above)
Comment (quiet-node) [reply test/yul/bitwise-coverage/Bitwise.js:24]: Ah for this I thought with this new batch of test suites for Yul support I just give it a new number. I believe before the Yul support, it went up to `solidityequiv4` so I just made it `solidityequiv5`

File: contracts/yul/bitwise-coverage/Bitwise.sol [hunk 2]
```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (Nana-EC) [inline contracts/yul/bitwise-coverage/Bitwise.sol:1]: Use Apache 2.0 for all licenses

**Review by Nana-EC — COMMENTED** [review]
LG
License fix

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**